l10n_it_account_balance_report/report/account_balance_report.py

```diff
@@ -14,58 +13,122 @@ def get_xmlid(id_str):
 
 class ReportAccountBalanceReport(models.TransientModel):
     _name = "account_balance_report"
-    _inherit = "account_financial_report_abstract"
-    _inherits = {"report_trial_balance": "trial_balance_id"}
+    _description = "Account balance report"
+    _inherit = "report.account_financial_report.abstract_report"
+
+    GROUP_TYPE = "group_type"
+    ACC_TYPE = "account_type"
 
     account_balance_report_type = fields.Selection(
         [("profit_loss", "Profit & Loss"), ("balance_sheet", "Balance Sheet")],
     )
+
+    company_id = fields.Many2one(
```

Comment:
Ho capito che tutte queste colonne vengono aggiunte perché non sono più recuperabili dal vecchio `trial_balance_id`, ma come mai è necessario aggiungerle? Non si può invece seguire la logica che viene utilizzata nel Trial Balance?
Ad esempio invece di dichiarare il campo `company_id` (che non viene più definito come campo in nessun report di `account_financial_report`), si potrebbe utilizzare il valore `tb_data['company_id']`.

In questo modo credo che le modifiche della migrazione sarebbero molto più ridotte, ma forse mi sfugge qualche problema in questo approccio?
In pratica, mi pare si stia cercando di ricreare la vecchia logica dei report `account_financial_report` invece di adattarsi alla nuova logica, e questo implica la creazione di tutti questi campi e le centinaia di righe di codice aggiunte nella migrazione.
Se possibile (come scrivevo, magari mi sfugge qualcosa per cui tutto ciò è necessario), cercherei di mantenere le modifiche di migrazione minimali.

Reply:
per i campi `company_id | date_from | date_to` sono stato costretto a lasciarli perché servono per la generazione del report.

Reply:
Ho provato a seguire la logica utilizzata dal Trial Balance in https://github.com/OCA/l10n-italy/pull/3343 e non serve aggiungere nuovi campi, fammi sapere cosa ne pensi